contracts/asset-proxy/contracts/src/bridges/BancorBridge.sol

```diff
@@ -0,0 +1,135 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "@0x/contracts-erc20/contracts/src/LibERC20Token.sol";
+import "@0x/contracts-exchange-libs/contracts/src/IWallet.sol";
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "../interfaces/IERC20Bridge.sol";
+import "../interfaces/IBancorNetwork.sol";
+
+
+contract BancorBridge is
+    IERC20Bridge,
+    IWallet,
+    DeploymentConstants
+{
+
+    IContractRegistry contractRegistry = IContractRegistry(0x52Ae12ABe5D8BD778BD5397F99cA900624CfADD4);
+    bytes32 bancorNetworkName = bytes32(uint(0x42616e636f724e6574776f726b)); // "BancorNetwork"
+
+    struct TransferState {
+        uint minReturn;
+        address[] addressPath;
+        IERC20Token[] tokenPath;
+    }
+
+    /// @dev Callback for `IERC20Bridge`. Tries to buy `amount` of
+    ///      `toTokenAddress` tokens by selling the entirety of the `fromTokenAddress`
+    ///      token encoded in the bridge data, then transfers the bought
+    ///      tokens to `to`.
+    /// @param toTokenAddress The token to buy and transfer to `to`.
+    /// @param from The maker (this contract).
+    /// @param to The recipient of the bought tokens.
+    /// @param amount Minimum amount of `toTokenAddress` tokens to buy.
+    /// @param bridgeData The abi-encoded conversion path addresses
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address toTokenAddress,
+        address from,
+        address to,
+        uint256 amount,
+        bytes calldata bridgeData
+    )
+        external
+        returns (bytes4 success)
+    {
+        // hold variables to get around stack depth limitations
+        TransferState memory state;
+
+        // Decode the bridge data.
+        (
+            state.addressPath,
+            state.minReturn
+        // solhint-disable indent
+        ) = abi.decode(bridgeData, (address[], uint));
+        // solhint-enable indent
+
+        require(state.addressPath.length >= 2, "BancorBridge/PATH_LENGTH_MUST_BE_AT_LEAST_TWO");
+        require(state.addressPath[state.addressPath.length - 1] == toTokenAddress, "BancorBridge/LAST_ELEMENT_OF_PATH_MUST_MATCH_OUTPUT_TOKEN");
+
+        // Just transfer the tokens if they're the same.
+        if (state.addressPath[0] == toTokenAddress) {
+            LibERC20Token.transfer(state.addressPath[0], to, amount);
+            return BRIDGE_SUCCESS;
+        }
+
+        // Otherwise use Bancor to convert
+        // Convert the addresses in the path to IERC20Tokens
+        for (uint i=0; i < state.addressPath.length; i++) {
```

Comment:
we should be able to avoid this by  changing the interface of Bancor Contracts to be `address` or `address[]` instead of `IERC20Token`